server/src/main/java/org/elasticsearch/rest/ChunkedRestResponseBody.java

```diff
@@ -132,6 +146,11 @@ public ReleasableBytesReference encodeChunk(int sizeHint, Recycler<BytesRef> rec
             public String getResponseContentTypeString() {
                 return builder.getResponseContentTypeString();
             }
+
+            @Override
+            public void setChunkedSizeListener(Consumer<Integer> sizeConsumer) {
+                this.sizeConsumer = sizeConsumer;
+            }
```

Comment:
Change to this class is to support getting response size for chunked responses. I have only just started look into these classes. Please let me know if there is a better way to achieve it. Thanks!